### PR TITLE
Fix/filter out manage neuron topic

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

According to [Flutter](https://github.com/dfinity/nns-dapp/pull/511)
- NeuronManagement proposals are not public so we hide this topic
- unless the neuron already has followees on this topic

# Changes

- refactoring to switch to followeesByTopic util function
- apply topicsToFollow to visible topics

# Tests

- followeesByTopic
- topicsToFollow

# Screens

## Before
image

## After
image

### with ManageNeuron followee

image

image

### CODE REVIEW

File: frontend/svelte/src/tests/lib/utils/neuron.utils.spec.ts [hunk 1]
```diff
@@ -879,4 +882,108 @@ describe("neuron-utils", () => {
...
+    it("should return topics without ManageNeuron", () => {
+      expect(topicsToFollow(neuronWithoutManageNeuron)).toEqual(
+        enumValues(Topic).filter((topic) => topic !== Topic.ManageNeuron)
+      );
+      expect(topicsToFollow(neuronWithoutFollowees)).toEqual(
+        enumValues(Topic).filter((topic) => topic !== Topic.ManageNeuron)
+      );
+    });
+
+    it("should return topics with ManageNeuron if already voted for it", () => {
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/tests/lib/utils/neuron.utils.spec.ts:985]: typo I guess, hehe. If neuron follows some neuron on the ManageNeuron topic.

**Review by lmuntanerDfinity — APPROVED** [review]
Thanks, LGTM, only a test description change.

Björn just mentioned to add this feature for heavy users: https://dfinity.slack.com/archives/C01S03NBM7S/p1650656373629389

File: frontend/svelte/src/lib/utils/neuron.utils.ts [hunk 2]
```diff
@@ -426,3 +428,28 @@ export const isIdentityController = ({
...
 };
+
+export const followeesByTopic = ({
+  neuron,
+  topic,
+}: {
+  neuron: NeuronInfo | undefined;
+  topic: Topic;
+}): Followees | undefined =>
+  neuron === undefined
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/utils/neuron.utils.ts:439]: nth: instead if the comparison why not `neuron?.fullNeuron...`

if neuron is undefined then it returns undefined too or am I missing something?

File: frontend/svelte/src/lib/utils/neuron.utils.ts [hunk 3]
```diff
@@ -426,3 +428,28 @@ export const isIdentityController = ({
...
+/**
+ * NeuronManagement proposals are not public so we hide this topic
+ * (unless the neuron already has followees on this topic)
+ * https://github.com/dfinity/nns-dapp/pull/511
+ */
+export const topicsToFollow = (neuron: NeuronInfo): Topic[] =>
+  enumValues(Topic).filter(
+    (topic) =>
+      topic !== Topic.ManageNeuron ||
+      followeesByTopic({ neuron, topic: Topic.ManageNeuron }) !== undefined
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/utils/neuron.utils.ts:454]: nth: for performance reason would maybe be better to assign `followeesByTopic` to a `const` before `filter`

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks. Two small nth comments

File: frontend/svelte/src/tests/lib/utils/neuron.utils.spec.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/svelte/src/tests/lib/utils/neuron.utils.spec.ts:985]: Good catch, thanks!

File: frontend/svelte/src/lib/utils/neuron.utils.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/utils/neuron.utils.ts]: Good point 👍 Updated .

File: frontend/svelte/src/lib/utils/neuron.utils.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/utils/neuron.utils.ts]: Relic. Removed.

**Pull request merged**